src/Sylius/Bundle/AddressingBundle/DependencyInjection/Compiler/RegisterZoneFactoryPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\AddressingBundle\DependencyInjection\Compiler;
+
+use Sylius\Bundle\AddressingBundle\Factory\ZoneFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class RegisterZoneFactoryPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sylius.factory.zone')) {
+            return;
+        }
+
+        $baseZoneFactory = $container->getDefinition('sylius.factory.zone');
+        $zoneFactoryDefinition = new Definition(ZoneFactory::class);
```

Comment:
You should use ``$baseZoneFactory->getClass();``. It will have the custom factory class in case it has been configured.